src/System.Private.ServiceModel/tests/Common/Scenarios/ScenarioTestTypes.cs

```diff
@@ -1104,3 +1106,35 @@ protected virtual void Dispose(bool disposing)
         disposed = true;
     }
 }
+
+public class MyX509CertificateValidator : X509CertificateValidator
+{
+    string allowedIssuerName;
+
```

Comment:
Can we put the `validateMethodWasCalled` here rather than have the static here? https://github.com/dotnet/wcf/pull/483/files#diff-5cc28940b9c4a251952d5274724a7f0fR25

Your test method has a reference to the instance of `MyX509CertificateValidator` used for validation, so it should be possible to keep this in an instance variable and validate the instance. 

The issue here is that if someone uses `MyX509CertificateValidator` without being aware of of its behaviour of the static in `ScenarioTestHelpers`, then you'll get some weird hard-to-debug test results.